src/app/modules/modal-shell/modal-shell.component.html

```diff
@@ -1,54 +1,61 @@
-    <div tabindex="0" class="focus-trap-hidden" id="firstModalTab" (focus)="findTab($event)" aria-hidden="true"></div>
-        <div class="tableClass">
-            <div class="cellClass">
-                <div class="ds-c-dialog {{modalClass}}" role="document">
-                    <app-button
-                        class="floatRight"
-                        (callBack)="onClose()" 
-                        [ariaLabel]="'Close modal dialog'" 
-                        [buttonType]="'modalTabIndex ds-c-button--transparent ds-c-dialog__close'" 
-                        [buttonID]="'closeButton'+modalCount">
-                        Close
-                    </app-button>
-                    
-                    <ng-template #defaultHeader let-modalHeader="modalHeader">
-                        <h1 class="ds-h1" id="{{'dialog-title-' + modalHeader.modalCount}}">
-                            {{modalHeader.modalTitle}}
-                        </h1>
-                    </ng-template>
-                    
-                    <ng-container *ngTemplateOutlet="headerTemplate ? headerTemplate : defaultHeader; context: getTitle">
-                    </ng-container>
+<div id="{{'modalImplementation' + modalCount}}" class="ds-c-dialog-wrap" [attr.aria-modal]="true"
+     [attr.aria-labelledby]="'dialog-title-' + modalCount" [attr.aria-describedby]="'dialog-body-' + modalCount"
+     role="dialog" id="modalDialog" id="modal" (keyDown.esc)="onClose()" attr.data-auto-id="{{dataAutoId}}">
+  <div tabindex="0" class="focus-trap-hidden" id="firstModalTab" (focus)="findTab($event)" aria-hidden="true"></div>
+  <div class="tableClass">
+    <div class="cellClass">
+      <div class="ds-c-dialog {{modalClass}}" role="document">
+        <div class="modal-header-wrapper ds-u-display--flex ds-u-margin-bottom--2">
+          <div class="modal-header-content">
+            <ng-container
+              *ngTemplateOutlet="headerTemplate ? headerTemplate : defaultHeader; context: getTitle">
+            </ng-container>
 
-                    <main [id]="jawsBody ? 'dialog-body-' + modalCount : ''">
+            <ng-template #defaultHeader let-modalHeader="modalHeader">
+              <h2 class="ds-h2"
+                  id="{{'dialog-title-' + modalHeader.modalCount}}">
+                {{modalHeader.modalTitle}}
+              </h2>
+            </ng-template>
+          </div>
+          <app-button
+            class="modal-header-close-button ds-u-margin-left--1"
+            (callBack)="onClose()"
+            [ariaLabel]="'Close modal dialog'"
+            [buttonType]="'modalTabIndex ds-c-button--transparent ds-c-dialog__close'"
+            [buttonID]="'closeButton'+modalCount">
+            Close
+          </app-button>
+        </div>
```

Comment:
Actual changes are here...(the rest is mostly just reformatting...sorry, it's a habit!)

Wrap the title and close button in .modal-header-wrapper
Change defaultHeader template to use h2 instead of h1